### PR TITLE
Abort run by default on SCF convergence failure

### PR DESCRIPTION
**Pull request by mkrack**
The new keyword IGNORE_CONVERGENCE_FAILURE can be set to obtain the old permissive behaviour. It replaces the keyword NOTCONV_STOPALL which enforced until now the strict behaviour which is now the default.

### CODE REVIEW

**Comment by oschuett** [comment]
Oh wow, how did you format all our test input files? Should we maybe make that a separate commit so we can later add it to [.git-blame-ignore-revs](https://github.com/cp2k/cp2k/blob/master/.git-blame-ignore-revs)?

**Comment by mkrack** [comment]
> Oh wow, how did you format all our test input files?

`vi` with some macro support.

> Should we maybe make that a separate commit so we can later add it to [.git-blame-ignore-revs](https://github.com/cp2k/cp2k/blob/master/.git-blame-ignore-revs)?

No objections.

**Comment by oschuett** [comment]
To separate out the actual changes you could try  `git format-patch --ignore-space-change ...`.

**Comment by mkrack** [comment]
> To separate out the actual changes you could try `git format-patch --ignore-space-change ...`.

Got two patch files ... ?

**Comment by oschuett** [comment]
I was thinking you could do something like this: 
1. Use `git format-patch --ignore-space-change` to save your manual modification as patch files.
2. Reset your branch.
3. Run your vim macros again.
4. Make a first commit, which contains only the formatting.
5. Apply the patch files.
6. Make a second commit, which contains the `NOTCONV_STOPALL` changes.

**Comment by mkrack** [comment]
Step 3 was not an automated process, because I still edited all files by hand.

**Comment by oschuett** [comment]
> Step 3 was not an automated process, because I still edited all files by hand.

Ok, I think I found a way. Please do not merge.

**Comment by mkrack** [comment]
> Ok, I think I found a way. Please do not merge.

@oschuett Thanks for the patch file. The test is clean now after some minor fixes. OK to merge?

**Comment by oschuett** [comment]
Yes, all good. This nicely solves a long standing usage pitfall.

File: src/qs_scf.F [hunk 1]
```diff
@@ -598,11 +598,12 @@ SUBROUTINE scf_env_do_scf(scf_env, scf_control, qs_env, converged, should_stop,
          dft_control%qs_control%cdft_control%total_steps = &
          dft_control%qs_control%cdft_control%total_steps + total_steps
 
-      IF (.NOT. converged) CPWARN("SCF run NOT converged")
```
Comment (oschuett) [inline src/qs_scf.F:601]: Maybe we should keep the old warning message? I'd expect it's hard coded in many places.

**Pull request merged**